pages/[...slug].vue

```diff
@@ -3,32 +3,41 @@
     <NuxtLayout :name="layout">
       <main>
         <SiteBreadcrumbs />
-        <DrupalTabs v-if="page.local_tasks" :tabs="page.local_tasks" />
-        <component :is="renderCustomElements(page.content)" />
+        <DrupalTabs v-if="page?.local_tasks" :tabs="page.local_tasks" />
+        <component :is="renderCustomElements(page.content)" v-if="page?.content"  />
       </main>
     </NuxtLayout>
   </div>
 </template>
 
 <script lang="ts" setup>
+import { useAsyncData, useRoute, useHead, computed } from '#imports'
+import { useDrupalCe } from '#imports'
+
+const route = useRoute()
 const { fetchPage, renderCustomElements } = useDrupalCe()
-const page = await fetchPage(useRoute().path, { query: useRoute().query })
-// Set to false to support custom layouts, using <NuxtLayout> instead.
+
+const { data: page } = await useAsyncData('page', () => 
```

Comment:
uhm, I'd be quite hesitate to change this - this is a quite a big change. could you explain why it's need / what it brings us?